progress/meter.go

```diff
@@ -106,29 +104,27 @@ func NewMeter(options ...meterOption) *ProgressMeter {
 
 // Start begins sending status updates to the optional log file, and stdout.
 func (p *ProgressMeter) Start() {
-	if atomic.CompareAndSwapInt32(&p.started, 0, 1) {
-		go p.writer()
-	}
+	atomic.StoreUint32(&p.paused, 0)
 }
 
 // Pause stops sending status updates temporarily, until Start() is called again.
 func (p *ProgressMeter) Pause() {
-	if atomic.CompareAndSwapInt32(&p.started, 1, 0) {
-		p.finished <- true
-	}
+	atomic.StoreUint32(&p.paused, 1)
 }
 
 // Add tells the progress meter that a single file of the given size will
 // possibly be transferred. If a file doesn't need to be transferred for some
 // reason, be sure to call Skip(int64) with the same size.
 func (p *ProgressMeter) Add(size int64) {
+	defer p.update()
```

Comment:
What's the benefit of using `defer` here?  Couldn't you just put `p.update()` at the end of the function?

Reply:
I don't think there's any semantic benefit to having this be a deferred function call, but I like that the `defer` keyword allows us to put the `p.update()` call near the top of the method as a sort of way to annotate "hey, this is going to produce some updates that will change the output of this on the terminal". Deferred calls are pretty fast nowadays (and this code is running in the context of uploading/downloading large files over the network, so I think that the performance concern would be negligible.

Reply:
Why not just leave the `defer` off then and keep it at the top? It looks like the only reason it'd hang is if the channel is full, in which case it's okay to delay the counter update.

Reply:
When `update()` is deferred, it reflects an update that includes the changes made by calling `Add()`, `Skip()`, etc. I think that it would be fine to run it before, we would just get something that says `Git LFS: 0 of 0 files ...`, and would have to `update()` once more when calling `Finish()`.

I don't feel strongly either way, so am happy to change it either way!